src/libs/kata-types/src/config/hypervisor/mod.rs

```diff
@@ -237,6 +239,14 @@ impl BootInfo {
         Ok(())
     }
 
+    /// Add kernel parameters to bootinfo. It is always added before the original
+    /// to let the original one takes priority
+    pub fn add_kparams(&mut self, params: Vec<String>) {
+        let mut p = params;
+        p.push(self.kernel_params.clone()); // [new_params0, new_params1, ..., original_params]
+        self.kernel_params = p.join(KERNEL_PARAM_DELIMITER);
+    }
```

Comment:
It would be helpful to check for errors somewhere in here.

Reply:
Sorry for the late reply. I will add some unit tests for `add_kparams` and `get_agent_kparams`, and add some error checking in `add_kparams`.

The reason why I didn't add error check here is that function here now only receives `params` from `KernelParam.to_string()`, where there is some error check.
But I think it would be safer to add an extra error check here.

Reply:
Fixed